# AG002 

##### SILAS RODRIGUES DE OLIVEIRA JUNIOR - GES 282

##### LUIZ GUSTAVO DE OLIVEIRA COSTA - GEC 1845

In [11]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import classification_report

# 1. Ler CSV
df = pd.read_csv('palmerpenguins.csv', sep=';')
df.head()

,species,island,culmen_length_mm,culmen_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181,3750,MALE
1,Adelie,Torgersen,39.5,17.4,186,3800,FEMALE
2,Adelie,Torgersen,40.3,18.0,195,3250,FEMALE
3,Adelie,Torgersen,36.7,19.3,193,3450,FEMALE
4,Adelie,Torgersen,39.3,20.6,190,3650,MALE


In [12]:
# 2. Padronizar nomes das colunas
df.columns = df.columns.str.strip().str.lower()
df.columns

Index(['species', 'island', 'culmen_length_mm', 'culmen_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex'],
      dtype='str')

In [13]:
# 3. Remover espaços em strings
for col in ['species', 'island', 'sex']:
    df[col] = df[col].astype(str).str.strip()
df[['species','island','sex']].head()

,species,island,sex
0,Adelie,Torgersen,MALE
1,Adelie,Torgersen,FEMALE
2,Adelie,Torgersen,FEMALE
3,Adelie,Torgersen,FEMALE
4,Adelie,Torgersen,MALE


In [14]:
# 4. Remover valores nulos
df = df.dropna()
df.isnull().sum()

species              0
island               0
culmen_length_mm     0
culmen_depth_mm      0
flipper_length_mm    0
body_mass_g          0
sex                  0
dtype: int64

In [15]:
df['island'] = df['island'].replace({
    'Biscoe': 0,
    'Dream': 1,
    'Torgersen': 2
})

df['sex'] = df['sex'].replace({
    'FEMALE': 0,
    'MALE': 1
})

df['species'] = df['species'].replace({
    'Adelie': 0,
    'Chinstrap': 1,
    'Gentoo': 2
})

df = df.astype({
    'island': 'int64',
    'sex': 'int64',
    'species': 'int64'
})


for c in ['island', 'sex', 'species']:
    print(f"\n{df[c].value_counts().sort_index()}")

print("\nTipos das colunas após conversão:")
print(df[['island', 'sex', 'species']].dtypes)


island
0    163
1    123
2     47
Name: count, dtype: int64

sex
0    165
1    168
Name: count, dtype: int64

species
0    146
1     68
2    119
Name: count, dtype: int64

Tipos das colunas após conversão:
island     int64
sex        int64
species    int64
dtype: object


In [16]:
# 6. Reordenar colunas
df = df[
    [
        'island',
        'sex',
        'culmen_length_mm',
        'culmen_depth_mm',
        'flipper_length_mm',
        'body_mass_g',
        'species'
    ]
]

print(df.columns.tolist())


['island', 'sex', 'culmen_length_mm', 'culmen_depth_mm', 'flipper_length_mm', 'body_mass_g', 'species']


In [23]:
# 7. Separar dados
X = df.drop('species', axis=1)
y = df['species']

# 8. Split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)
print("\nColunas:")
print(X.columns.tolist())

print("\nDimensões:")
print(f"- X: {X.shape}")
print(f"- y: {y.shape}")
print(f"- X_train: {X_train.shape}")
print(f"- X_test: {X_test.shape}")
print(f"- y_train: {y_train.shape}")
print(f"- y_test: {y_test.shape}")



Colunas:
['island', 'sex', 'culmen_length_mm', 'culmen_depth_mm', 'flipper_length_mm', 'body_mass_g']

Dimensões:
- X: (333, 6)
- y: (333,)
- X_train: (266, 6)
- X_test: (67, 6)
- y_train: (266,)
- y_test: (67,)


In [18]:
# 9. Modelo
model = DecisionTreeClassifier(random_state=42)
# 10. Treinar
model.fit(X_train, y_train)
print('Treinamento concluído')

Treinamento concluído


In [19]:
# 11. Testar
y_pred = model.predict(X_test)
# 12. Métricas
print('\nRelatório de Classificação:\n')
print(classification_report(y_test, y_pred))


Relatório de Classificação:

              precision    recall  f1-score   support

           0       1.00      0.97      0.98        31
           1       0.93      1.00      0.96        13
           2       1.00      1.00      1.00        23

    accuracy                           0.99        67
   macro avg       0.98      0.99      0.98        67
weighted avg       0.99      0.99      0.99        67



In [20]:
# 13. Classificação manual

print('--- Classificação manual ---')
island = int(input('Island (0=Biscoe, 1=Dream, 2=Torgersen): '))
sex = int(input('Sex (0=FEMALE, 1=MALE): '))
culmen_length = float(input('Culmen length (mm): '))
culmen_depth = float(input('Culmen depth (mm): '))
flipper_length = float(input('Flipper length (mm): '))
body_mass = float(input('Body mass (g): '))

# Criar DataFrame com nomes de colunas para evitar warning
entrada = pd.DataFrame([[island, sex, culmen_length, culmen_depth, flipper_length, body_mass]], columns=X_train.columns)
pred = model.predict(entrada)[0]
mapa = {0: 'Adelie', 1: 'Chinstrap', 2: 'Gentoo'}
print('Especie prevista:', mapa[pred])

# Probabilidades por classe
probs = model.predict_proba(entrada)[0]
classes = model.classes_
probs_por_classe = { mapa[int(c)]: float(p) for c, p in zip(classes, probs) }
print('Probabilidades por classe:', probs_por_classe)

--- Classificação manual ---
Especie prevista: Adelie
Probabilidades por classe: {'Adelie': 1.0, 'Chinstrap': 0.0, 'Gentoo': 0.0}


## TESTES (Adelie):

Island: 0

Sex: 0

Culmen length (mm): 37.8

Culmen depth (mm): 18.3

Flipper length (mm): 174

Body mass (g): 3400


## TESTE 2 (Chinstrap):


Island: 1

Sex: 1

Culmen length (mm): 50.0

Culmen depth (mm): 19.0

Flipper length (mm): 195

Body mass (g): 3700



## TESTE 3 (Gentoo):


Island: 0

Sex: 1

Culmen length (mm): 48.5

Culmen depth (mm): 14.5

Flipper length (mm): 215

Body mass (g): 5200